doc/platform_api/CMIS_Diagnostic_Monitoring_Overview_in_SONiC.md

```diff
@@ -0,0 +1,2375 @@
+Ethernet0    Tx CDR LOL on host Lane 7        False/  0/  never/  never
+Ethernet0    Tx CDR LOL on host Lane 8        False/  0/  never/  never
+Ethernet0    Tx EQ fault on host Lane 1       False/  0/  never/  never
+Ethernet0    Tx EQ fault on host Lane 2       False/  0/  never/  never
+Ethernet0    Tx EQ fault on host Lane 3       False/  0/  never/  never
+Ethernet0    Tx EQ fault on host Lane 4       False/  0/  never/  never
+Ethernet0    Tx EQ fault on host Lane 5       False/  0/  never/  never
+Ethernet0    Tx EQ fault on host Lane 6       False/  0/  never/  never
+Ethernet0    Tx EQ fault on host Lane 7       False/  0/  never/  never
+Ethernet0    Tx EQ fault on host Lane 8       False/  0/  never/  never
+Ethernet0    Rx CDR LOL on media Lane 1       False/  0/  never/  never
+Ethernet0    Rx CDR LOL on media Lane 2       False/  0/  never/  never
+Ethernet0    Rx CDR LOL on media Lane 3       False/  0/  never/  never
+Ethernet0    Rx CDR LOL on media Lane 4       False/  0/  never/  never
+Ethernet0    Rx CDR LOL on media Lane 5       False/  0/  never/  never
+Ethernet0    Rx CDR LOL on media Lane 6       False/  0/  never/  never
+Ethernet0    Rx CDR LOL on media Lane 7       False/  0/  never/  never
+Ethernet0    Rx CDR LOL on media Lane 8       False/  0/  never/  never
+Ethernet0    Target output power out of range False/  0/  never/  never
+Ethernet0    Fine tuning out of range flag    False/  0/  never/  never
+Ethernet0    Tuning not accepted flag         False/  0/  never/  never
+Ethernet0    Invalid channel number flag      False/  0/  never/  never
+Ethernet0    Tuning complete flag             False/  0/  never/  never
+```
+
+## 4. SONiC CMIS diagnostic monitoring workflow
+
+### 4.1 Static Diagnostic Information
+
+The `SfpStateUpdateTask` thread is responsible for updating the static diagnostic information for all the transceivers in the system. The static diagnostic information, such as threshold values for DOM, VDM and PM, are read from the transceiver and updated in the `redis-db` during `xcvrd` boot-up and during transceiver removal and insertion.
+
+The following tables are updated by the `SfpStateUpdateTask` thread:
+
+1. `TRANSCEIVER_DOM_THRESHOLD`
+2. `TRANSCEIVER_VDM_XXX_THRESHOLD` where `XXX` is the threshold type (`highalarm`, `highwarning`, `lowwarning`, `lowalarm`)
+
+### 4.2 Dynamic Diagnostic Information
+
+The `DomInfoUpdateTask` thread is responsible for updating the dynamic diagnostic information for all the transceivers in the system. The following events drive the dynamic update of the diagnostic information:
+
+1. **Periodic update of the diagnostic information:**
+    - The `DomInfoUpdateTask` thread periodically updates the diagnostic information for all the ports.
+    - The update period interval can be retrieved by reading the `dom_info_update_periodic_secs` field from the `/usr/share/sonic/device/{platform}/{hwsku}/pmon_daemon_control.json` file.
```

Comment:
@mihirpat1 is it ok to have .json per platform? Reason being Xcvrd will do DOM polling per physical port count which is same across all HWSKUs (so far) and unlike polling of all logical ports

Reply:
@prgeor Agreed - I have removed the hwsku from the path now.